pulp_2_tests/tests/rpm/api_v2/test_modularity.py

```diff
@@ -156,85 +146,191 @@ def remove_module_from_repo(self, repo, criteria):
+
+    def test_copy_modulemd_defaults_recursive_conservative(self):
+        """Test copy of modulemd_defaults in RPM repository."""
+        repo = self.copy_units(True, True, self.criteria)
+        self.check_module_total_units(repo)
+
+    def check_module_total_units(self, repo):
+        """Test copy of modulemd_defaults in RPM repository."""
         self.assertEqual(
-            repo['content_unit_counts']['rpm'],
-            MODULE_FIXTURES_PACKAGE_STREAM['rpm_count'],
-            repo['content_unit_counts']['rpm']
+            repo['content_unit_counts']['modulemd_defaults'],
+            MODULE_FIXTURES_PACKAGE_STREAM['module_defaults'],
+            repo['content_unit_counts']
         )
         self.assertEqual(
             repo['total_repository_units'],
-            MODULE_FIXTURES_PACKAGE_STREAM['total_available_units'],
+            MODULE_FIXTURES_PACKAGE_STREAM['module_defaults'],
             repo['total_repository_units']
         )
+        self.assertNotIn('rpm', repo['content_unit_counts'])
 
-    def test_copy_modulemd_non_recur(self):
-        """Test copy of modulemd in RPM repository in non recursive mode."""
-        criteria = {
+    def copy_units(self, recursive, recursive_conservative, criteria, old_dependency=False):
+        """Create two repositories and copy content between them."""
+        repos = []
+        body = gen_repo(
+            importer_config={'feed': RPM_WITH_MODULES_FEED_URL},
+            distributors=[gen_distributor()]
+        )
+        repos.append(self.client.post(REPOSITORY_PATH, body))
+        self.addCleanup(self.client.delete, repos[0]['_href'])
+        sync_repo(self.cfg, repos[0])
+        repos.append(self.client.post(REPOSITORY_PATH, gen_repo()))
+        self.addCleanup(self.client.delete, repos[1]['_href'])
+        # Add `old_dependency` for OLD RPM on B
+        if old_dependency:
+            rpm = utils.http_get(RPM_WITH_OLD_VERSION_URL)
+            upload_import_unit(
+                self.cfg,
+                rpm,
+                {'unit_type_id': 'rpm'}, repos[1]
+            )
+            units = search_units(self.cfg, repos[1], {'type_ids': ['rpm']})
+            self.assertEqual(len(units), 1, units)
+
+        self.client.post(urljoin(repos[1]['_href'], 'actions/associate/'), {
+            'source_repo_id': repos[0]['id'],
+            'override_config': {
+                'recursive': recursive,
+                'recursive_conservative': recursive_conservative,
+            },
+            'criteria': criteria
+        })
+        return self.client.get(repos[1]['_href'], params={'details': True})
+
+
+class CopyModulesTestCase(unittest.TestCase):
+    """Test copy of modules, and its artifacts.
+
+    This test targets the following issues:
+
+    * `Pulp Smash #1122 <https://github.com/PulpQE/pulp-smash/issues/1122>`_
+    * `Pulp #4543 <https://pulp.plan.io/issues/4543>`_
+
+    Modules and RPM packages used in this test case are them
```

Comment:
```suggestion
    Modules and RPM packages used in this test case are the
```

Not sure the intend here, just double checking.